zipline-gradle-plugin/src/main/kotlin/app/cash/zipline/gradle/ZiplineCompiler.kt

```diff
@@ -77,10 +78,32 @@ object ZiplineCompiler {
         }
       }
     }
-    val manifest = ZiplineManifest.create(modules)
+
+    val sortedModules = ZiplineManifest.create("temp", "temp", modules).modules
+    val applicationModuleJsFileNameWithoutExtention = sortedModules.entries.last().key.removePrefix("./").removeSuffix(".js")
+    val outputTsTypeDefinitionFile = File(inputDir.path, "${applicationModuleJsFileNameWithoutExtention}.d.ts")
+    val applicationId = "./${applicationModuleJsFileNameWithoutExtention}.js"
+    val prepareFunction = getPrepareFunctionName(outputTsTypeDefinitionFile)
+
+    val manifest = ZiplineManifest.create(
+      applicationId = applicationId,
+      prepareFunction = prepareFunction,
+      modules = modules,
+    )
     val manifestFile = File(outputDir.path, "manifest.zipline.json")
     manifestFile.writeText(Json.encodeToString(manifest))
   }
+
+  internal fun getPrepareFunctionName(tsTypeDefinitionFile: File): String {
+    val contents = tsTypeDefinitionFile.readText()
+    return getPrepareFunctionName(contents)
+  }
+
+  internal fun getPrepareFunctionName(contents: String): String {
```

Comment:
I don’t think we can do this generally. . . For example, this is the activity.d.ts file:

```
type Nullable<T> = T | null | undefined
declare const __doNotImplementIt: unique symbol
type __doNotImplementIt = typeof __doNotImplementIt
export namespace com.squareup.cash.js {
    class WebScriptsConfig {
        constructor(cashAppBaseUrl: string);
        get statementsBaseUrl(): string;
        get refundableWindowDays(): number;
        get paymentUrlBase(): string;
        get rightToRefundLinks(): any/* kotlin.js.Json */;
        get squareAddressUrl(): string;
        get helpCenterLinks(): any/* kotlin.js.Json */;
        get documentsBaseUrl(): string;
        get installmentPlanUrl(): string;
        get cryptoOrderReceiptUrl(): string;
    }
}
export namespace com.squareup.cash.js {
    class PaymentHistoryApi {
        constructor();
        getOfflinePaymentHistoryDataJson(serializedInitiatePaymentRequest: string, serializedAppVersion: string, serializedClientData: Nullable<string>): string;
        getPaymentHistoryDataJson(serializedPayment: string, serializedSender: string, serializedRecipient: string, serializedAppVersion: string, serializedClientData: Nullable<string>, serializedReceipt: Nullable<string>, serializedCheckDepositReceipt: Nullable<string>): string;
        getSerializedOfflinePaymentHistoryData(): any;
        getSerializedPaymentHistoryData(): any;
    }
}
export namespace com.squareup.cash.js.lib {
    const Logger: {
        log(message: string): void;
        warn(message: string): void;
        error(message: string): void;
    };
}
export namespace com.squareup.cash.treehouse.activity {
    function prepareTreehouse(): void;
}
export as namespace payment_history_treehouse;
```

Reply:
What if we required a naming convention for these? Like, for example, the function name needed to start with `prepareTreehouse` ? We’d have to document it but it’d be automatic-enough

Reply:
Another option that’s a bit more work . . . We could create our own annotation, `@PrepareTreehouse`

Reply:
Agreed, I was suspicious that there'd be a more complex Treehouse repo that broke my brittle RegEX parsing. 

Some initial looking at annotations seems a bit tricky since the current Zipline compiler is only accepting JS files, not the Kotlin src, so we don't get any reflection. I can look into adding src code reflection if that would be best.

Alternatively, I could add a parameter to the ZiplineCompile task where the developer needs to specify the main function explicitly and have it match their code.